### PR TITLE
Modernize `DQM/BeamMonitor/plugins/TKStatus` plugin

### PR DESCRIPTION
**Pull request by francescobrivio**
#### PR description:
In the same spirit of #29198, #35782 and #37084, this PR modernizes the `DQM/BeamMonitor/plugins/TKStatus` plugin to use `DCSRecord` token in case the DCS information from SCAL is not available (i.e. from Run3 onwards).
Changes:
 - unified `TKStatus.cc` and `TKStatus.h`
 - added `DCSRecord` consumption (and a logic to choose between it and `DCSstatus`)
 - added `process.onlineMetaDataDigis` unpacker in the only DQM client running in production that uses TkStatus
 - added `process.onlineMetaDataDigis` unpacker to a bunch of configs under `DQM/BeamMonitor/test` that were importing and using TkStatus

#### PR validation:
Code compiles.
Tested by running on run `348490`:
```
cmsRun DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py unitTest=True dataset='/ExpressCosmics/Commissioning2022-Express-v1/FEVT' runNumber=348490
```
which has only Pixels ON.
And by running on run `348587`:
```
cmsRun DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py unitTest=True dataset='/ExpressCosmics/Commissioning2022-Express-v1/FEVT' runNumber=348587
```
which has both Pixel and Strips ON.

#### Backport
A backport to 12_3_X is provided in #37223
A backport to 12_2_X is provided in #37224

### CODE REVIEW

**Comment by mmusich** [comment]
@francescobrivio 
you are missing the `onlineMetaDataDigis` unpacker from the executed paths (see https://github.com/cms-sw/cmssw/pull/30677/).
The reason why you don't have them for the DQM test is because - in order to mimic the real data-taking scenario in which you don't run over FEVT but rather on un-reconstructed streamer files - I drop the non-`FEDRawData` branches here:

https://github.com/cms-sw/cmssw/blob/523b3608a975de3078708f3319e928f0e9e1acc0/DQM/Integration/python/config/unittestinputsource_cfi.py#L119-L124

precisely to catch this sort of issues upstream and not when the client is run online at P5 :)

File: DQM/BeamMonitor/plugins/TKStatus.cc [hunk 1]
```diff
@@ -1,50 +1,112 @@
...
-      if (!status.ready(DcsStatus::TECm))
-        dcsTk[5] = false;
+    // Check that the DCS information is available in some form
+    if (!dcsStatus.isValid() && !dcsRecord.isValid()) {
+      edm::LogPrint("TkStatus") << "No DCS information available!";
+      dumpTkDcsStatus(dcsTkFileName_, iEvent.run(), dcsTk);
     }
+
+    // If DCS status (from SCAL) not available (or empty), use DCS record
+    if ((*dcsStatus).empty()) {
```
Comment (mmusich) [inline DQM/BeamMonitor/plugins/TKStatus.cc:84]: what about the MC case?
If there's a MC case...

File: DQM/BeamMonitor/plugins/TKStatus.cc [hunk 2]
```diff
@@ -1,50 +1,112 @@
...
+  void analyze(const edm::Event& e, const edm::EventSetup& c) override;
 
-TKStatus::TKStatus(const ParameterSet& ps) {
-  dcsTkFileName_ = ps.getParameter<ParameterSet>("BeamFitter").getUntrackedParameter<std::string>("DIPFileName");
+private:
+  void dumpTkDcsStatus(std::string const&, edm::RunNumber_t, std::array<bool, 6> const&);
+
+  std::string dcsTkFileName_;
+  edm::EDGetTokenT<DcsStatusCollection> dcsStatusToken_;
+  edm::EDGetTokenT<DCSRecord> dcsRecordToken_;
```
[the remark this answers is not in the document]
Comment (francescobrivio) [reply DQM/BeamMonitor/plugins/TKStatus.cc]: foxed in [fab8b2b](https://github.com/cms-sw/cmssw/pull/37214/commits/fab8b2b56bf7116096e3dcf9da32bae8e500653a)

File: DQM/BeamMonitor/plugins/TKStatus.cc [hunk 3]
```diff
@@ -1,50 +1,112 @@
...
   int nthlumi = iEvent.luminosityBlock();
   if (nthlumi > lastlumi_) {  // check every LS
     lastlumi_ = nthlumi;
     // Checking TK status
-    Handle<DcsStatusCollection> dcsStatus;
-    iEvent.getByToken(dcsStatus_, dcsStatus);
+    edm::Handle<DcsStatusCollection> dcsStatus;
+    iEvent.getByToken(dcsStatusToken_, dcsStatus);
+    edm::Handle<DCSRecord> dcsRecord;
+    iEvent.getByToken(dcsRecordToken_, dcsRecord);
```
[the remark this answers is not in the document]
Comment (francescobrivio) [reply DQM/BeamMonitor/plugins/TKStatus.cc]: fixed in [fab8b2b](https://github.com/cms-sw/cmssw/pull/37214/commits/fab8b2b56bf7116096e3dcf9da32bae8e500653a)

File: DQM/BeamMonitor/plugins/TKStatus.cc (hunk 1 above)
Comment (francescobrivio) [reply DQM/BeamMonitor/plugins/TKStatus.cc]: I improved the overall checking logic in a similar way of #37084

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
@cms-sw/dqm-l2 do we also need a 12_2 backport? 
In principle beamspot clients are not important for cosmics, but if you want to test it in the playback system just let me know and I can open a 12_2 backport.

**Comment by francescobrivio** [comment]
> @cms-sw/dqm-l2 do we also need a 12_2 backport? In principle beamspot clients are not important for cosmics, but if you want to test it in the playback system just let me know and I can open a 12_2 backport.

I opened a 12_2_X backport in #37224

**Comment by jfernan2** [comment]
+1
Tested successfully in Online DQM at P5 (no crashes).

**Comment by qliphy** [comment]
+1

**Pull request merged**